common_crawler/main.py

```diff
@@ -119,10 +123,40 @@ def main():
             keyword=args.keyword,
             last_page=common_crawl_result.last_page_search)
         cache_manager.save_cache()
+
     except ValueError as e:
         print(f"Error while saving cache manager: {e}")
 
 
+def validate_remote_results(remote_results):
+    """
+    Validates the remote results retrieved from the Label Studio project
+
+    Args: remote_results (dict or list): The results from the Label Studio project
+
+    Returns:
+        list[dict]: If the remote results are valid
+        None: If the remote results are invalid
+    """
+    if isinstance(remote_results, list):
+        if not remote_results:
+            print("No data in Label Studio project.")
+            return []
+        elif "url" not in remote_results[0]["data"]:
+            sys.exit("Column 'url' not present in Label Studio project. Exiting...")
+        else:
+            return remote_results
+    elif isinstance(remote_results, dict):
+        if remote_results.get("status_code") == 401:
+            sys.exit("Invalid Label Studio token passed! Exiting...")
+        elif remote_results.get("status_code") == 404:
+            sys.exit("Invalid Label Studio project ID! Exiting...")
+        else:
+            sys.exit(f"Unexpected error: {remote_results}")
+    else:
+        sys.exit("Unexpected response type.")
```

Comment:
Two things here:
1. As with before, recommend refactor to avoid the arrow-head anti-pattern. For example: the `elif isinstance(remote_results, dict):` block is all about handling different errors, so that can be extracted to a `handle_remote_results_error` function (or whatever name you think is appropriate).
2. Using sys.exit on errors makes things harder to test and can be less clear: they imply what is occurring is part of the expected logic path rather than an exceptional situation. If errors are occurring, replace this with raising errors, either custom errors or standard Python errors.